cmake/std/atdm/ATDMDevEnvSettings.cmake

```diff
@@ -438,8 +455,16 @@ ENDIF()
 ATDM_SET_ENABLE(TPL_ENABLE_ParMETIS ${TPL_ENABLE_ParMETIS_DEFAULT})
 IF (TPL_ENABLE_ParMETIS)
   ASSERT_DEFINED(ENV{PARMETIS_ROOT})
-  ATDM_SET_CACHE(ParMETIS_INCLUDE_DIRS "$ENV{PARMETIS_ROOT}/include" CACHE FILEPATH)
-  ATDM_SET_CACHE(ParMETIS_LIBRARY_DIRS "$ENV{PARMETIS_ROOT}/lib" CACHE FILEPATH)
+
+  # if the naming in ATDM_CONFIG matched cmake variables, this is alot cleaner
+  ATDM_SET_ATDM_VAR_FROM_ENV_AND_DEFAULT(PARMETIS_INCLUDE_DIRS
+					 "$ENV{PARMETIS_ROOT}/include")
```

Comment:
@e10harvey @bartlettroscoe (consider this)

I'm going to digress a bit in this post - but it's totally worth you and Evan thinking about.

Ross, I've been using this snipplet to unset ATDM stuff for while.  What do you think replacing the bulk of that file with
```
for atdm_var in $(compgen -A variable | grep '^ATDM_CONFIG'); do 
  # for example only
  echo "Before: $atdm_var=${!atdm_var}"; 

  # the real deal
  unset $atdm_var;

  # for example only
  echo "After: $atdm_var=${!atdm_var}";
done

# the variables are actually undefined afterwards... but I print the name, so `ha=`
Before: ATDM_CONFIG_ha=1
After: ATDM_CONFIG_ha=
Before: ATDM_CONFIG_he=1
After: ATDM_CONFIG_he=
Before: ATDM_CONFIG_lol=3
After: ATDM_CONFIG_lol=
```

In the context of ATDMDevSettings, you could even write this to create a cache VAR for every `ATDM_CONFIG_<foo...>`

I've toyed with this (and it works wonderfully).

In a refactor, you'd move all of the non-prefixed `ATDM_CONFIG` variables to be `ATDM_CONFIG_<name>`. This would allow an `atdm author` to arbitrarily inject necessary variables.

The logic in ATDMDevSettings then gets cleaner. You effectively check for variables that are set (and define some policy/precedence for whether `find` is used vs raw `TPL_FOO_LIBRARIES` is used.

The code below would allow you to disable BOOST, by doing `ATDM_CONFIG_TPL_ENABLE_BOOST=OFF`
A little polish to recognize  `types`, but this would really simplify the soup.

```
function(load_atdm_config_variables)
# run a basic bash command grab all variable **names** in the ENV
# restrict to starting with ATDM_CONFIG_
# strip the ATDM_CONFIG_ prefix
# then package the result into list
#
# Note, these are variables NAMES, not values
execute_process(COMMAND bash "-c" "compgen -A variable"
                COMMAND grep "^ATDM_CONFIG"
                COMMAND sed "s/^ATDM_CONFIG_//"
                COMMAND tr  "[:space:]" ";"
                RESULT_VARIABLE rc
                OUTPUT_VARIABLE thebomb)

message( "rc == ${rc}" )
message( "${thebomb}" )
set(atdm_names ${thebomb})
message(${atdm_names})
foreach(e ${thebomb})
  message("ATDM_CONFIG_${e} = $ENV{ATDM_CONFIG_${e}} ")
  set(${e} $ENV{ATDM_CONFIG_${e}} CACHE STRING "user ATDM_CONFIG variable" FORCE)
endforeach()

endfunction()
```


Effectively what I am proposing is that ATDMDev take `ownership` of all ENVs prefixed `ATDM_CONFIG` - that seems perfectly reasonable to me - Then, in an `environment.sh` file, we **only** operate on `ATDM_CONFIG` prefixed variables.

This would avoid name clashes... e.g., `MPICC` or really anything.

The top-most snipplet shows how to en-masse remove ATDM_CONFIG variables, and the bottom shows how Cmake could be made to create variables from it (you could twiddle it to be CACHE - or use `macro` - and I think you could set a file local variable instead (probably preferred!).